lib/HttpClient/CurlClient.php

```diff
@@ -147,6 +147,15 @@ public function request($method, $absUrl, $headers, $params, $hasFile)
             throw new Error\Api("Unrecognized method $method");
         }
 
+        // It is only safe to retry network failures on post and delete
+        // requests if we add an Idempotency-Key header
+        if ((($method == 'post') || ($method == 'delete')) &&
+            (Stripe::$maxNetworkRetries > 0)) {
+            if (!isset($headers['Idempotency-Key'])) {
+                $headers['Idempotency-Key'] = bin2hex(openssl_random_pseudo_bytes(32));
+            }
+        }
+
```

Comment:
I think you meant "PUT" here and not "DELETE"?
[Your own documentation](https://stripe.com/docs/api#idempotent_requests) says it's not necessary for DELETE.

Reply:
I cargo-culted the implementation from stripe-ruby, but you're absolutely correct: idempotency does not apply to DELETE requests.

The API does not use PUT requests at all, so POST is the only method that can use idempotency. I'll update the PR. Thanks 👍

Reply:
Don't forget to update the comment too.